plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DriverConnectionFactory.java

```diff
@@ -73,7 +73,7 @@ public Connection openConnection(ConnectorSession session)
         JdbcIdentity identity = JdbcIdentity.from(session);
         Properties properties = getCredentialProperties(identity);
         Connection connection = driver.connect(connectionUrl.apply(identity), properties);
-        checkState(connection != null, "Driver returned null connection");
+        checkState(connection != null, "Driver returned null connection, make sure the connection URL '%s' is valid for the driver %s", connectionUrl, driver);
```

Comment:
I am not sure what to log here -- the `connectionUrl` or `connectionUrl.apply(identity)`.

Reply:
`connectionUrl` is a `Function`. So not a very loggable thing.

Reply:
You're right. I was too quick.

Reply:
I will chat with @kokosing why we need a function here.

Reply:
I think it is no longer used. And could be removed. I haven't found any usages of it.

Reply:
Thanks @kokosing for checking! will simplify